packages/flutter/lib/src/painting/borders.dart

```diff
@@ -258,14 +296,15 @@ class BorderSide {
     return other is BorderSide
         && other.color == color
         && other.width == width
-        && other.style == style;
+        && other.style == style
+        && other.strokeAlign == strokeAlign;
   }
 
   @override
-  int get hashCode => Object.hash(color, width, style);
+  int get hashCode => Object.hash(color, width, style, strokeAlign);
 
   @override
-  String toString() => '${objectRuntimeType(this, 'BorderSide')}($color, ${width.toStringAsFixed(1)}, $style)';
+  String toString() => '${objectRuntimeType(this, 'BorderSide')}($color, ${width.toStringAsFixed(1)}, $style, $strokeAlign)';
```

Comment:
I think you can avoid printing the `, $strokeAlign` if it is set to `inside`.  This is consistent with how `Diagnosticable` handles default values, and would keep it from invalidating all the existing tests.

Reply:
Fixed with an `if`, like `Diagnosticable`.